### PR TITLE
Expose TTLClockGen for Kasli JSONs

### PR DESCRIPTION
**Pull request by Spaqin**
# ARTIQ Pull Request

## Description of Changes

TTLClockGen module is available already, but it's not exposed in JSONs. While the class is used in KC705-based systems, but for Kasli it was not yet possible to have DIO clock generation available. This PR only allows usage of clockgen with kasli systems.

Tested by running a simple ARTIQ experiment using the clockgen and looking at the output on oscilloscope.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :sparkles: New feature |

### All Pull Requests

- [x] Use correct spelling and grammar.

### Code Changes

- [x] Run `flake8` to check code style (follow PEP-8 style). `flake8` has issues with parsing Migen/gateware code, ignore as necessary.
- [x] Test your changes or have someone test them. Mention what was tested and how.

### Git Logistics

- [x] Split your contribution into logically separate changes (`git rebase --interactive`). Merge/squash/fixup commits that just fix or amend previous commits. Remove unintended changes & cleanup. See [tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
- [x] Write short & meaningful commit messages. Review each commit for messages (`git show`). Format:
  ```
  topic: description. < 50 characters total.
  
  Longer description. < 70 characters per line
  ```

### Licensing

See [copyright & licensing for more info](https://github.com/m-labs/artiq/blob/master/CONTRIBUTING.rst#copyright-and-sign-off).
ARTIQ files that do not contain a license header are copyrighted by M-Labs Limited and are licensed under LGPLv3+.

### CODE REVIEW

File: artiq/gateware/rtio/phy/ttl_simple.py [hunk 1]
```diff
@@ -145,11 +145,18 @@ def __init__(self, pad):
...
+    def __init__(self, pad, pad_n=None, ftw_width=24, dci=False):
         self.rtlink = rtlink.Interface(rtlink.OInterface(ftw_width))
 
         # # #
 
+        pad_o = Signal()
+        if pad_n is None:
+            self.comb += pad.eq(pad_o)
+        else:
+            self.specials += Instance("IOBUFDS",
```
Comment (sbourdeauducq) [inline artiq/gateware/rtio/phy/ttl_simple.py:157]: Use ``DifferentialOutput`` and remove redundant code in ``eem.py`` (``Urukul.add_std``).

**Comment by sbourdeauducq** [comment]
Needs RELEASE_NOTES update.
I'm also not sure how useful that is considering that all four channels would become clock outputs.
And anyway that feature is low priority I think.

**Comment by Spaqin** [comment]
I had it done to test NAC3 in a nice way anyway, so no harm done in adding it to mainline I think. Or if it's no good then also no problem either, as usefulness is limited as you mentioned...

File: artiq/gateware/eem.py [hunk 2]
```diff
@@ -231,10 +231,8 @@ def add_std(cls, target, eem, eem_aux, ttl_out_cls, sync_gen_cls=None, iostandar
         target.rtio_channels.append(rtio.Channel.from_phy(phy, ififo_depth=4))
 
         pads = target.platform.request("urukul{}_dds_reset_sync_in".format(eem))
-        pad = Signal(reset=0)
-        target.specials += DifferentialOutput(pad, pads.p, pads.n)
         if sync_gen_cls is not None:  # AD9910 variant and SYNC_IN from EEM
-            phy = sync_gen_cls(pad, ftw_width=4)
+            phy = sync_gen_cls(pad=pads.p, pad_n = pads.n, ftw_width=4)
```
Comment (sbourdeauducq) [inline artiq/gateware/eem.py:235]: style

**Pull request merged**